### ISSUE TITLE
Refactor

### ISSUE THREAD

**Issue by timbocode** [ClassicPress-research/classic-forms-8#1]
Change:

- text domain to `classic-forms-8`
- `Contact Form 7` and `CF7` to `Classic Forms 8` or `CF8`
- `WPCF7_` to `CPCF8_`
- `wpcf7` to `cpcf8`

Please add any others below.

**Comment by bahiirwa** [comment]
I have some reservations on changing the names of the classes and methods in the plugin. Many of these are “hookable” like `WPCF7_Submission` 

A number of extensions that use these classes will be unusable. Can we adopt this like we did with ClassicCommerce?

This particular plugin has a number of extensions in its ecosystem. My cents. 

If we however go the new route, what’s the plan for those plugins?

**Comment by timbocode** [comment]
@bahiirwa You raise a good point or two but I don't have any definitive answers.

When I first suggested forking CF7, in the longer term, I envisaged it being a base for our own forms plugin rather than just a compatible copy of CF7. 

The problem we're going to increasingly come up against with any plugin we fork (including CC) is that extensions will also stop being compatible with CP and we'll be left with no option but to fork those too. This applies whether we "adopt" it like we did with CC or whether we completely refactor it. The only difference, as I see it, is timing.

## Adopting
### Pros
Relatively quick and easy with a readily accessible bunch of extensions / addons without needing to fork them just yet.

### Cons
Although I haven't looked into this, if we adopt CF7, it's likely that we'd also need to develop a compatibility plugin as we've done with CC. But this means that for every plugin we adopt, users may also need to install a compatibility plugin. Things could get very unwieldy.

## Refactoring
### Pros
No need for a compatibility plugin and the plugin becomes "our own" from the start.

### Cons
If we want to take advantage of extensions, we'd need to fork (and refactor) them upfront.

So that's the dilemma. We need to get James's view.

**Comment by bahiirwa** [comment]
> But this means that for every plugin we adopt, users may also need to install a compatibility plugin. Things could get very unwieldy.

I do not think we want to keep doing this. 

> the plugin becomes "our own" from the start.
If we are going for the bare minimums. This is ok for now. It reduces the baggage.

**Comment by ginsterbusch** [comment]
The refactor goes a bit too far. I'd like to keep it as compatible to the main CF 7 stable branch as possible. Instead, I'd suggest keeping the existing constants etc., but enhancing them with its own, including functions / class methods for version + compatiblity checks. Ie. similar to how its implemented within ClassicPress.

cu, w0lf.

**Comment by timbocode** [comment]
OK. 

For version 1, there are two main aims:

- no compatibility plugin
- 100% compatibility with CF7 and extensions

If we can work within that, then all good with me.

**Comment by nylen** [comment]
> The refactor goes a bit too far. I'd like to keep it as compatible to the main CF 7 stable branch as possible. Instead, I'd suggest keeping the existing constants etc., but enhancing them with its own, including functions / class methods for version + compatiblity checks. Ie. similar to how its implemented within ClassicPress.

Yep, this is in line with my thinking also. An additional "con" to changing things now, especially class names and function names, is that a bunch of extensions will definitely stop working, and there is also a fairly high risk of breaking something in the plugin code itself because sometimes people do crazy things like dynamically building strings and using them to reference a variable/class/method by name.

The presence or absence of a compatibility plugin seems to be a separate issue. If it's found that CF7 extensions check for `is_plugin_active: contact-form-7` (pseudocode because I don't remember the exact snippet that many Woo extensions were using) then I kind of think it's unavoidable, at least that was the conclusion reached from early work on Classic Commerce.
